tests/src/test/scala/whisk/core/cli/test/WskWebActionsTests.scala

```diff
@@ -230,6 +230,25 @@ trait WskWebActionsTests
             response.header("Access-Control-Allow-Headers") shouldBe "Headers set from Web Action"
     }
 
+    it should "ensure that CORS header is preserved for non-custom options" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "webaction"
+            val file = Some(TestUtils.getTestActionFilename("corsHeaderMod.js"))
+
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) =>
+                    action.create(name, file, web = Some("true"))
+            }
+
+            val host = getServiceURL()
+            val url = host + s"$testRoutePath/$namespace/default/webaction.http"
```

Comment:
that is for both .http and .json, the controller should be adding the expected headers for access control. 
furthermore, let's make sure we're not overriding and removing other headers and the test action should have other headers included.